IntroComponentWithSign-upForm/script.js

```diff
@@ -0,0 +1,67 @@
+const form  = document.getElementById("form")
+const firstName  = document.getElementById("firstName")
+const lastName  = document.getElementById("lastName")
+const email  = document.getElementById("email")
+const password  = document.getElementById("password")
+//show error message
+const showErrorMessage = ( input , message ) => {
+    const span = document.querySelector("."+input.id)  
+    console.log(span.innerText = message);
+}
+// Show input success message
+const showSuccessMessage = ( input ) => {
+    const span = document.querySelector("."+input.id)
+    span.innerText = "";
+}
+//get field name
+const getFieldName = (input) =>{
+    return input.name;   
+} 
+// check required fields
+const checkRequired = (inputArr) => {    
+    inputArr.forEach( (input)=>{ 
+        if ( input.value.trim() === "")
+        { 
+            showErrorMessage(input , getFieldName(input) +"  " + "cannot be empty")
+             //icon error    
+             var erIcon = document.querySelectorAll(".iconError")
+             for (var i = 0; i < erIcon.length; i++) {
+                erIcon[i].style.visibility = "visible";;
+             }    
+            // place holder email color
+           var El = getFieldName(input)      
+           if (El === "email"){
+             var emailEl = document.getElementById("email")
+                emailEl.classList.add("emailInvalidInput");
+                emailEl.placeholder='email@example.com';
+           } else{
+            document.getElementsByName('first Name')[0].placeholder='';
+            document.getElementsByName('last Name')[0].placeholder='';
+            document.getElementsByName('password')[0].placeholder='';
```

Comment:
Here, I'd recommend for you to use `getElementById` instead of `getElementsByName`. First, because an `id` in HTML, should never be duplicated, so it's easier to find the exact element that you are looking for, and second, the input field's name might change (even dynamically in the future) so this way, your code is way more scalable!

And one last thing, the return of the `getElementsById` is the element itself, not a `NodeArray`, so it's simpler and easier to read as well! 😄 

http://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById

```suggestion
            document.getElementsById('firstName').placeholder = '';
            document.getElementsById('lastName').placeholder = '';
            document.getElementsById('password').placeholder = '';
```